### ISSUE TITLE
Question: Data shuffling

### ISSUE THREAD

**Issue by L-Reichardt** [mlfoundations/model-soups#2]
I have found this Paper very interesting and have applied the method. I've notice the following:
- Greedy soup increase performance when souping from the same run, e.g. epoch 59+60 = 0.3% increase
- Souping from different runs with different augmentations, I get worse results sometimes -60%.

In the Paper you mention Frankle et al. (2020) and their observation that data order matters. Did you find a correlation here to greedy soup, e.g. that data shuffling reduces results?

I have to do more testing on my side, currently limited by the time to generate new weights. Also, I have trained all model so far from scratch instead of fine tuning, however I plan to test with fine tuned models later on.

**Comment by mitchellnw** [comment]
Hello and sorry for the delay!

- Averaging weights from along the same trajectory is done in EMA of SWA (e.g., https://arxiv.org/abs/1803.05407). It's interesting to hear that you can do greedy soup on top of that.
- If you're not in the fine-tuning context I do not expect soups to work. Indeed, our experiments are in the setting where you start with a model that is already performing well then fine-tune it.

With regards to data order: whenever we make soups from identical hyperparameters, the data order changing is what makes the models different.

**Comment by L-Reichardt** [comment]
@mitchellnw

Thanks a lot for your answer. That clears up all the questions I had.

Regarding EMA / SWA : I used a different benchmark / network, which does not use EMA / SWA. I just tested the algorithm to make sure it works.
